handlers.go

```diff
@@ -33,14 +33,31 @@ type Handler interface {
 	Handle(ctx context.Context, call *InboundCall)
 }
 
-// A HandlerFunc is an adapter to allow the use of ordering functions as
-// Channel handlers.  If f is a function with the appropriate signature,
-// HandlerFunc(f) is a Hander object that calls f
+// A HandlerFunc is an adapter to allow the use of ordinary functions as
+// Channel handlers.  If f is a function with the appropriate signature, then
+// HandlerFunc(f) is a Handler object that calls f.
 type HandlerFunc func(ctx context.Context, call *InboundCall)
 
 // Handle calls f(ctx, call)
 func (f HandlerFunc) Handle(ctx context.Context, call *InboundCall) { f(ctx, call) }
 
+// An ErrorHandlerFunc is an adapter to allow the use of ordinary functions as
+// Channel handlers, with error handling convenience.  If f is a function with
+// the appropriate signature, then ErrorHandlerFunc(f) is a Handler object that
+// calls f.
+type ErrorHandlerFunc func(ctx context.Context, call *InboundCall) error
+
+// Handle calls f(ctx, call)
+func (f ErrorHandlerFunc) Handle(ctx context.Context, call *InboundCall) {
+	if err := f(ctx, call); err != nil {
+		call.log.WithFields(ErrField(err)).Error("Handler error.")
```

Comment:
here and below, we should add the handlerr name (which should be accessible via `call`) so the error log has more information